src/torchmetrics/functional/detection/panoptic_quality.py

```diff
@@ -282,22 +343,20 @@ def _panoptic_quality_compute(
         false_negatives: the FN value from the update step
 
     Returns:
-        panoptic quality
+        Panoptic quality as a tensor containing a single scalar.
     """
-    # TODO: exclude from mean categories that are never seen ?
-    # TODO: per class metrics
-
     # per category calculation
     denominator = (true_positives + 0.5 * false_positives + 0.5 * false_negatives).double()
     panoptic_quality = torch.where(denominator > 0.0, iou_sum / denominator, 0.0)
-    return torch.mean(panoptic_quality)
+    # Reduce across categories. TODO: is it useful to have the option of returning per class metrics?
+    return torch.mean(panoptic_quality[denominator > 0])
```

Comment:
Note: this is a change in behaviour of the metric. Intuitively it does make sense to me, but we can argue whether there should be an option of masking out never seen classes or not

Reply:
Fine by me :)